### PR TITLE
Feature/Newsfeed update-after git pull origin main 2/3/2022

### PR DESCRIPTION
**Pull request by pjaepole**
## Description
- more changes to newsfeed for both instructor and parent
- redux added only for fetching newsfeeds, other inputfields and such still have local states
- form validation added for post request
- added modal pop up for individual news(styling still needed)
- both instructor and parent newsfeed now use our backend data instead of dummy data
- separate form for parent, instructor(instructor's form have functions to edit delete, but parent's form only have the option to view)
- few CSS className conflicts and bugs fixed
- axioswithauth added to all axios requests
- check https://github.com/BloomTech-Labs/coder-heroes-fe/pull/134 for more commit history

Fixes # (issue)

## Loom Video

Please paste a link here of a quick loom video walking through what you did in your PR .

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [ ] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/components/pages/InstructorNewsFeed/NewsContainer.js [hunk 1]
```diff
@@ -2,41 +2,26 @@ import React, { useEffect } from 'react';
...
 import {getNewsFeeds} from '../../../redux/actions/instructorActions';
+import IndividualNews from './IndividualNews';
 function NewsContainer(props) {
   const { setPostId, setPostOptions, newsfeed, dispatch }=props;
   useEffect(() => {
-    const token = JSON.parse(localStorage.getItem('okta-token-storage'));
-    const config = {
-      headers: { Authorization: `Bearer ${token.idToken.value}` },
-    };
-    dispatch(getNewsFeeds(config));
```
Comment (JesusNavarrete215) [inline src/components/pages/InstructorNewsFeed/NewsContainer.js:12]: Removing the token authentication cleans up the code and using the axiosWithAuth from utils will make everything  easy to read.

File: src/components/pages/ParentNewsFeed/NewsContainer.js [hunk 2]
```diff
@@ -1,55 +1,35 @@
...
-            </div>
-            <div className="description">
-              <h3>{description}</h3>
-            </div>
-            <div className="button-container">
-              <a href={link}>Link</a>
-            </div>
-          </div>
-        );
-      })}
```
Comment (JesusNavarrete215) [inline src/components/pages/ParentNewsFeed/NewsContainer.js:52]: I like the code being refactored to include state and removing the unnecessary anchors.

**Review by JesusNavarrete215 — COMMENTED** [review]
I like how you removed and refactored your code to make everything a bit easier to follow. Using the axiosWithAuth is beneficial to making your get, post, and delete requests.

**Review by CamMeiners — APPROVED** [review]
I like the implementation of axiosWithAuth, it's also very cool to see all the Redux research you guys have done come to life.

**Review by ElchinNasirov — APPROVED** [review]

**Review by manny-glez — APPROVED** [review]
Great work, it Looks great.

**Pull request merged**